exchanges/kline/kline.go

```diff
@@ -349,8 +350,29 @@ func (k *Item) ConvertToNewInterval(newInterval Interval) (*Item, error) {
 	if len(candles) == 0 {
 		return nil, fmt.Errorf("%w to %v no candle data", ErrInsufficientCandleData, newInterval)
 	}
+
 	var target int
 	for x := range k.Candles {
+		if k.Candles[x].Open == 0 &&
+			k.Candles[x].High == 0 &&
+			k.Candles[x].Low == 0 &&
+			k.Candles[x].Close == 0 &&
+			k.Candles[x].Volume == 0 {
+			// This candle is padding, there is nothing to apply to the new
+			// interval candle. We can skip this candle and move on to the next.
```

Comment:
All being zero is not necessarily a padding candle. Some exchanges do return zero data when they go down for example (my memory says Binance does this). If this is specifically about padding you may wish to add a bool to kline to signify that it is padding rather than zero data